src/org/omegat/core/data/ITMXEntry.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * *************************************************************************
+ *  OmegaT - Computer Assisted Translation (CAT) tool
+ *           with fuzzy matching, translation memory, keyword search,
+ *           glossaries, and translation leveraging into updated projects.
+ *
+ *  Copyright (C) 2021 Hiroshi Miura.
+ *                Home page: http://www.omegat.org/
+ *                Support center: https://omegat.org/support
+ *
+ *  This file is part of OmegaT.
+ *
+ *  OmegaT is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  OmegaT is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *  *************************************************************************
+ *
+ */
+
+package org.omegat.core.data;
+
+import org.omegat.util.TMXProp;
+
+public interface ITMXEntry {
+    String getSource();
+
+    boolean isTranslated();
+
+    String getTranslation();
+
+    String getChanger();
+
+    long getChangeDate();
+
+    String getCreator();
+
+    long getCreationDate();
+
+    String getNote();
+
+    boolean hasNote();
+
+    boolean hasProperties();
+
+    String getPropValue(String propType);
+
+    boolean hasPropValue(String propType, String propValue);
+
+    Iterable<TMXProp> getProperties();
```

Comment:
Should expose `List<TMXProp>` instead of `Iterable<>` ? or provide `getRawProperties` method?

Reply:
Personally in interfaces I prefer to use Iterable instead of List. I have users who do not hesitate to use gigabytes of translation memories in a project, for that reason I have some implementations of my plugins which do not really load everything in memory, and for that reason sometimes the implementation of such a method is not necessarily a collection.

Reply:
That said, TMXEntry does not store properties and there is a reason: if you look in project_save.tmx you will see always the same properties which are stored in the entry key, not in the entry itself. So, no reason to have these methods in the interface (unless you see in the code agorithms which are using the properties and could be used for both chasses: in such a case, implementation in InternalTMXEntry should always be null)